### DISCUSSION TITLE
E2AP decoding question using Wireshark

### DISCUSSION THREAD

**Discussion by alexjiao2021** [srsran/srsRAN_Project#177] [General Help] [answered]
Hi, I tried E2 interface function following the guide:
https://docs.srsran.com/projects/project/en/latest/tutorials/source/flexric/source/index.html
I used Wireshark 4.0.7 to decode E2AP messages, some fields can't be decoded (shown as 'Unknown') as in below pic:
![image](https://github.com/srsran/srsRAN_Project/assets/84055495/fb6fbf58-5bcc-420e-b34f-fc3020c9f371)
Can someone please help? Thanks.
pcap file:
[gnb_e2ap.zip](https://github.com/srsran/srsRAN_Project/files/12387240/gnb_e2ap.zip)

**Comment by pgawlowicz** [comment, marked as answer]
Hi, thanks for your feedback. E2AP is still under development in Wireshark.
Probably the best option is to compile Wireshark from [source code.](https://github.com/wireshark/wireshark)
I am currently using version ``4.1.0 (v4.1.0rc0-3390-g4f4a54e6d3f9)`` compiled from the ``master`` branch, and it can decode the pointed part of the ``Setup Request`` message.
